### PR TITLE
[ru] Add RU localization - logging.md

### PR DESCRIPTION
**Pull request by kirkonru**
The following file was translated into Russian:

- content/ru/docs/concepts/cluster-administration/logging.md

### CODE REVIEW

File: content/ru/docs/concepts/cluster-administration/logging.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+...
+```
+
+Команда `kubectl logs --previous` позволяет извлечь логи из предыдущего воплощения контейнера. Если в Pod'е несколько контейнеров, выбрать нужный для извлечения логов можно с помощью флага `-c`:
+
+```console
+kubectl logs counter -c count
+```
+
+Для получения дополнительной информации см. [документацию по `kubectl logs`](/docs/reference/generated/kubectl/kubectl-commands#logs) for more details.
```
Comment (shurup) [inline content/ru/docs/concepts/cluster-administration/logging.md:61]:
```suggestion
Для получения дополнительной информации см. [документацию по `kubectl logs`](/docs/reference/generated/kubectl/kubectl-commands#logs).
```

File: content/ru/docs/concepts/cluster-administration/logging.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+Подробную информацию о том, как `kube-up.sh` настраивает логирование для образа COS на GCP, можно найти в соответствующем скрипте [`configure-helper`](https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/gci/configure-helper.sh).
+
+При использовании **среды исполнения контейнера CRI** kubelet отвечает за ротацию логов и управление структурой их директории. kubelet передает данные среде исполнения контейнера CRI, а та сохраняет логи контейнера в указанное место. С помощью параметров [`containerLogMaxSize` и `containerLogMaxFiles`](/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration) в [конфигурационном файле kubelet'а](/docs/tasks/administer-cluster/kubelet-config-file/) можно настроить максимальный размер каждого лог-файла и максимальное число таких файлов для каждого контейнера, соответственно.
```
Comment (shurup) [inline content/ru/docs/concepts/cluster-administration/logging.md:79]:
```suggestion
При использовании **среды исполнения контейнера CRI** kubelet отвечает за ротацию логов и управление структурой их директории. kubelet передает данные среде исполнения контейнера CRI, а та сохраняет логи контейнера в указанное место. С помощью параметров [`containerLogMaxSize` и `containerLogMaxFiles`](/docs/reference/config-api/kubelet-config.v1beta1/#kubelet-config-k8s-io-v1beta1-KubeletConfiguration) в [конфигурационном файле kubelet'а](/docs/tasks/administer-cluster/kubelet-config-file/) можно настроить максимальный размер каждого лог-файла и максимальное число таких файлов для каждого контейнера соответственно.
```

File: content/ru/docs/concepts/cluster-administration/logging.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+Сбор логов на уровне узла предусматривает запуск одного агента на узел и не требует изменений в приложениях, работающих на узле.
+
+Контейнеры пишут в `stdout` и `stderr`, но без согласованного формата. Агент на уровне узла собирает эти логи и направляет их на агрегацию.
+
+### Сбор логов с помощью sidecar-контейнера с лог-агентом {#sidecar-container-with-logging-agent}
+
+Sidecar-контейнер можно использовать одним из следующих способов:
+
+* sidecar-контейнер транслирует логи приложений на свой собственный `stdout`;
+* в sidecar-контейнере работает агент, настроенный на сбор логов из контейнера приложения..
```
Comment (shurup) [inline content/ru/docs/concepts/cluster-administration/logging.md:123]:
```suggestion
* в sidecar-контейнере работает агент, настроенный на сбор логов из контейнера приложения.
```

File: content/ru/docs/concepts/cluster-administration/logging.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+#### Транслирующий sidecar-контейнер
+
+![Sidecar-контейнер с транслирующим контейнером](/images/docs/user-guide/logging/logging-with-streaming-sidecar.png)
+
+Настроив sidecar-контейнеры на вывод в их сосбственные потоки `stdout` и `stderr`, можно воспользоваться преимуществами kubelet и лог-агента, которые уже работают на каждом узле. Sidecar-контейнеры считывают логи из файла, сокета или journald. Затем каждый из них пишет логи в собственный поток `stdout` или `stderr`.
```
Comment (shurup) [inline content/ru/docs/concepts/cluster-administration/logging.md:129]:
```suggestion
Настроив sidecar-контейнеры на вывод в их собственные потоки `stdout` и `stderr`, можно воспользоваться преимуществами kubelet и лог-агента, которые уже работают на каждом узле. Sidecar-контейнеры считывают логи из файла, сокета или journald. Затем каждый из них пишет логи в собственный поток `stdout` или `stderr`.
```

File: content/ru/docs/concepts/cluster-administration/logging.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+
+Обратите внимание: несмотря на низкое использование процессора и памяти (порядка нескольких milliCPU для процессора и пары мегабайт памяти), запись логов в файл и их последующая потоковая передача в `stdout` может вдвое увеличить нагрузку на диск. Если приложение пишет в один файл, рекомендуется установить `/dev/stdout` в качестве адресата, нежели использовать подход с транслирующим контейнером.
+
+Sidecar-контейнеры также можно использовать для ротации файлов логов, которые не могут быть ротированы самим приложением. В качестве примера можно привести небольшой контейнер, периодически запускающий `logrotate`. Однако рекомендуется использовать `stdout` и `stderr` напрямую, а управление политиками ротации и хранения оставить kubelet'у.
+
+#### Sidecar-контейнер с лог-агентом
+
+![Sidecar-контейнер с лог-агентом](/images/docs/user-guide/logging/logging-with-sidecar-agent.png)
+
+Если лог-агент на уровне узла недостаточно гибок для ваших потребностей, можно создать sidecar-контейнер с отдельным лог-агентом, специально настроенный на работу с приложением.
```
Comment (shurup) [inline content/ru/docs/concepts/cluster-administration/logging.md:181]:
```suggestion
Если лог-агент на уровне узла недостаточно гибок для ваших потребностей, можно создать sidecar-контейнер с отдельным лог-агентом, специально настроенным на работу с приложением.
```

**Review by shurup — CHANGES_REQUESTED** [review]
Small corrections & suggestions to consider.

File: content/ru/docs/concepts/cluster-administration/logging.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+weight: 60
+---
+
+<!-- overview -->
+
+Логи помогают понять, что происходит внутри приложения. Они особенно полезны для отладки проблем и мониторинга деятельности кластера. У большинства современных приложений имеется тот или иной механизм сбора логов. Контейнерные движки в этом смысле не исключение. Самый простой и наиболее распространенный метод сбора логов для контейнерных приложений задействует потоки `stdout` и `stderr`.
+
+Однако встроенной функциональности контейнерного движка или среды исполнения обычно недостаточно для организации полноценного решения по сбору логов.
+
+Например, может возникнуть необходимость просмотреть логи приложения при аварийном завершении работы Pod'а, его выселении (eviction) или "падении" узла.
```
[the remark this answers is not in the document]
Comment (kirkonru) [reply content/ru/docs/concepts/cluster-administration/logging.md:14]: Okay, I'll keep an eye on that, thanks for the suggestion.

File: content/ru/docs/concepts/cluster-administration/logging.md (hunk 1 above)
Comment (kirkonru) [reply content/ru/docs/concepts/cluster-administration/logging.md:61]: Thanks for spotting it.

File: content/ru/docs/concepts/cluster-administration/logging.md (hunk 6 above)
Comment (kirkonru) [inline content/ru/docs/concepts/cluster-administration/logging.md:14]:
```suggestion
Например, может возникнуть необходимость просмотреть логи приложения при аварийном завершении работы Pod'а, его вытеснении (eviction) или "падении" узла.
```

File: content/ru/docs/concepts/cluster-administration/logging.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+
+![Сбор логов на уровне узла](/images/docs/user-guide/logging/logging-node-level.png)
+
+Среда исполнения для контейнера обрабатывает и перенаправляет любой вывод в потоки `stdout` и `stderr` приложения. Docker Engine, например, перенаправляет эти потоки [драйверу журналирования](https://docs.docker.com/engine/admin/logging/overview), который в Kubernetes настроен на запись в файл в формате JSON.
+
+{{< note >}}
+JSON-драйвер Docker для сбора логов рассматривает каждую строку как отдельное сообщение. В данном случае поддержка многострочных сообщений отсутствует. Обработка многострочных сообщений должна выполняться на уровне лог-агента или выше.
+{{< /note >}}
+
+По умолчанию, если контейнер перезапускается, kubelet сохраняет один завершенный контейнер с его логами. Если Pod выселяется с узла, все соответствующие контейнеры также выселяются вместе с их логами.
```
Comment (shurup) [inline content/ru/docs/concepts/cluster-administration/logging.md:73]: "Eviction" still has an old translation here (twice), let's make it consistent.

Comment (kirkonru) [inline content/ru/docs/concepts/cluster-administration/logging.md:73]:
```suggestion
По умолчанию, если контейнер перезапускается, kubelet сохраняет один завершенный контейнер с его логами. Если Pod вытесняется с узла, все соответствующие контейнеры также вытесняются вместе с их логами.
```

Comment (kirkonru) [reply content/ru/docs/concepts/cluster-administration/logging.md:73]: Oh gosh, sorry, somehow I missed it.

**Comment by shurup** [comment]
@kirkonru we have [the same issue](https://github.com/kubernetes/website/pull/36328#issuecomment-1318023184) here. The files we need in this PR are:
* `debug/counter-pod.yaml`
* `admin/logging/two-files-counter-pod.yaml`
* `admin/logging/two-files-counter-pod-streaming-sidecar.yaml`
* `admin/logging/fluentd-sidecar-config.yaml`
* `admin/logging/two-files-counter-pod-agent-sidecar.yaml`

**Comment by shurup** [comment]
@kirkonru: a small step is required to finalise this PR as well.

/remove-lifecycle stale

**Comment by kirkonru** [comment]
@shurup hi, I added the required files.

**Comment by shurup** [comment]
@kirkonru great, thanks! Can you also squash all your commits, please?

**Comment by kirkonru** [comment]
@shurup commits are squashed. Could you please merge this PR?

**Pull request merged**